### DISCUSSION TITLE
Why is my password considered "weak" and cannot be used?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22118] [New to GitHub]
<p>I cannot set my desired password for my GitHub account, because it is considered “weak” and so I’m not allowed to use it.</p>
<p>But it’s strong enough according to the rules:</p>
<blockquote>
<pre><code>**Eight characters long, if it includes a number and a lowercase letter**, or
16 characters long with any combination of characters
</code></pre>
</blockquote>
<p>My password has 9 characters, one number and several lowercase and uppercase letters.</p>
<p>Even if this would be considered a weak password, why does the information above states that 8 characters, one number, and one lowercase letter should be enough?</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>Hi there! :wave: Welcome to the Community!</p>
<p>Our password policy is dictated both by the composition of the password (as indicated above) but we also do check against datasets of passwords that have been published.</p>
<p>In short, while your password may pass basic compositional validation, we know that the password you’re trying to use is not only vulnerable to attack but has already been included in datasets used by malicious actors to gain access to others’ accounts.</p>
<p>I hope that explains why we’d like you to use something else!</p>

**Comment by ghost** [comment]
<p>Thanks for the explanation. I did not know that my password was published. According to <a href="https://haveibeenpwned.com/Passwords" rel="nofollow noopener">https://haveibeenpwned.com/Passwords</a> it has been seen once.</p>
<p>Then I will choose another one.</p>
